contracts/SwapMigrator.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/ISwap.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+/**
+ * @title SwapMigrator
+ * @notice This contract is responsible for migrating old USD pool liquidity to the new ones.
```

Comment:
Is this specific to USD pools, or can this be used for btc / eth pools as well?

Reply:
This contract is only for USD pool but new ones can be easily created from this. The reason why we don’t have generalized swapmigrator is
* we need to know the relevant information before users can use the contract because we need to approve the tokens beforehand. Approving all tokens per user call would add a lot of gas for each user. 
* having generalized migrator means an owner needs to maintain the list of migrations. That seems like unnecessary work/added centralized risk for one-time migration feature that we may or may not use in the future